### PR TITLE
account for cell based stack frames, re-enable IW error output tests

### LINKED ISSUE #11306: HREF error traceback tests are failing
image

### PR DESCRIPTION
**Pull request by amunger**
Fixes #11306

old
"**Input** \u001b[1;32m**In [1]**\u001b[0m, in \u001b[0;36m**<cell line: 1>**\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124merror\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",

new
"**Cell** \u001b[1;32m**In [5], line 1**\u001b[0m\n\u001b[1;32m----> 1\u001b[0m \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124merror\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",

Creating the links correctly for notebooks is more difficult but has quite a few issues with it already (e.g. it always points to the current cell if it came from any cell)

### CODE REVIEW

**Comment by amunger** [comment]
Updated Ipykernel is in the lower cell. The cell URI isn't listed, but it's wrong in the current version if the code came from any other cell.

image

File: src/notebooks/outputs/tracebackFormatter.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export class NotebookTracebackFormatter implements ITracebackFormatter {
         return traceback.map((traceFrame) => this.modifyTracebackFrameIPython(cell, traceFrame));
     }
     private modifyTracebackFrameIPython(cell: NotebookCell, traceFrame: string): string {
-        if (/^[Input|File].*?\n.*/.test(traceFrame)) {
+        if (/^[Cell|Input|File].*?\n.*/.test(traceFrame)) {
```
Comment (IanMatthewHuff) [inline src/notebooks/outputs/tracebackFormatter.ts:26]: It looks like this code path also need the Cell regex. Or was the intention to just fix for IW first?

Comment (amunger) [reply src/notebooks/outputs/tracebackFormatter.ts:26]: adding cell here still gets into the more generic formatting that we do, just skips getting the cell URI

**Comment by amunger** [comment]
https://github.com/microsoft/vscode-jupyter/issues/11004 is the issue for cell URIs not being corrrect

**Review by IanMatthewHuff — APPROVED** [review]
Seems good as long as we are tracking the work to possibly fix the notebook output links as needed.

**Pull request merged**